python/pyspark/taskcontext.py

```diff
@@ -95,3 +95,92 @@ def getLocalProperty(self, key):
         Get a local property set upstream in the driver, or None if it is missing.
         """
         return self._localProperties.get(key, None)
+
+
+class BarrierTaskContext(TaskContext):
+
+    """
+    .. note:: Experimental
+
+    A TaskContext with extra info and tooling for a barrier stage. To access the BarrierTaskContext
+    for a running task, use:
+    L{BarrierTaskContext.get()}.
+
+    .. versionadded:: 2.4.0
+    """
+
+    _barrierContext = None
+
+    def __init__(self):
+        """Construct a BarrierTaskContext, use get instead"""
+        pass
```

Comment:
I would throw an exception BTW if this method should rather be banned

Reply:
This just follows `TaskContext.__init__()`, shall we update both?

Reply:
Ah, this is called in `_getOrCreate`. Sorry, I rushed to read. In this case, frankly I think we can remove this since that's the default constructor injected by Python or monkey patch to disallow the initialization (like we did for `ImageSchema`) but I guess we don't necessarily be super clever on this.

Reply:
I'm okay as is.